### ISSUE TITLE
In cassandra 2.1, nodetool repair -st / -et doesn't pass the "column family" option

### ISSUE THREAD

**Issue by Skunnyk** [BrianGallew/cassandra_range_repair#31]
On 2.0.x, I usually run repair with ``range_repair -k keyspace -c columnfamily``, which run : 

```
 nodetool repair -st (start token) -et (end token) $keyspace $columnfamily
```

The cassandra log was usually something like : 
```
StorageService.java (line 2496) starting user-requested repair of range [(300005707039417874,300156110175996167]] for keyspace $keyspace and column families [$columnfamily]
```

In 2.1, nodetool doesn't pass the "column family" (cfnames) option (as seen in the code in NodeTool.java), in case of use of startToken/endToken, so you repair all tables in the range… : See https://git-wip-us.apache.org/repos/asf?p=cassandra.git;a=blob;f=src/java/org/apache/cassandra/tools/NodeTool.java;h=1d4a42024dd406f0c90fc6b99f599b4ab4a0948e;hb=refs/heads/cassandra-2.1#l1917

```
if (!startToken.isEmpty() || !endToken.isEmpty())
    probe.forceRepairRangeAsync(System.out, keyspace, parallelismDegree, dataCenters,hosts, startToken, endToken, !incrementalRepair);
else
    probe.forceRepairAsync(System.out, keyspace, parallelismDegree, dataCenters, hosts, primaryRange, !incrementalRepair, cfnames);
```

And the log : 
```
StorageService.java:2846 - starting user-requested repair of range [(9187601611781349802,9189958813088431561]] for keyspace $keyspace and column families []
```

The "column family" is emtpy, so we are reparing all column families between theses 2 tokens.

Is it a bug (in cassandra ?), is it expected, or am I missing something ? :)

**Comment by BrianGallew** [comment]
That seems to be an exceptionally silly bug if it really works that way.  I
have never used 2.1 and so am unfamiliar with it.

**Comment by Skunnyk** [comment]
Fyi, it has been fixed in 2.1.16 (not yet released) :  https://issues.apache.org/jira/browse/CASSANDRA-11866 .

Thank you :)
